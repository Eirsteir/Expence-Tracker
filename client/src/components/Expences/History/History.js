import React from 'react';
import Paper from '@material-ui/core/Paper';
import { Line } from 'react-chartjs-2';

import MonthlyExpencesExpantionPanel from '../../ExpantionPanel/MonthlyExpencesExpantionPanel';
import Switch from '../../Switch/Switch';

import '../Expences.css';

const chartOptions = {
  maintainAspectRatio: false,
  scales: {
    yAxes: [{
      ticks: {
        suggestedMin: 10,
        suggestedMax: 100,
        fontColor: "rgba(0,0,0,0.5)",
        fontStyle: "bold",
        beginAtZero: true,
        maxTicksLimit: 5,
        padding: 20
      },
      gridLines: {
          drawTicks: false,
          display: false
      }
    }],
    xAxes: [{
      gridLines: {
        zeroLineColor: "transparent"
      },
      ticks: {
        padding: 20,
        fontColor: "rgba(0,0,0,0.5)",
      },

    }]
  },
  animation: {
    easing: 'easeOutExpo'
  },
  legend: {
    position: "bottom"
  },
}


class ExpenceHistory extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      checkedA: true,
      checkedB: false,
    };
  }

  handleChange = name => event => {
    if (name === 'checkedA') {
      return this.setState({
        checkedA: event.target.checked,
        checkedB: !event.target.checked,
      })
    } else if (name === 'checkedB') {
      return this.setState({
        checkedA: !event.target.checked,
        checkedB: event.target.checked,
      })
    }
  };


  chartYearlyData = canvas => {
    const now = new Date();
    // linear-gradient(89deg, #9fb8ad 0%, #1fc8db 51%, #2cb5e8 75%)
    const ctx = canvas.getContext("2d")
    const gradient = ctx.createLinearGradient(0,10,1000,20);
    gradient.addColorStop(0, 'rgba(	23, 105, 170, 0.6)');
    gradient.addColorStop(1, 'rgba(	0, 150, 136, 0.6)');

    const borderGradient = ctx.createLinearGradient(0,10,1000,20);
    borderGradient.addColorStop(0, '#1769aa');
    borderGradient.addColorStop(1, '#009688');

    return {
      labels: this.props.months.slice(0, now.getMonth() + 1),
      datasets: [
        {
          label: 'Amount',
          data: Object.keys(this.props.expencesSortedByMonth).map((month, i) => this.props.expencesSortedByMonth[month][1].total),
          fill: true,
          backgroundColor: gradient,
          lineTension:0.0000001,
          borderColor: borderGradient,
          borderWidth: 5,
          borderCapStyle: 'butt',
          borderDash: [],
          borderDashOffset: 0.0,
          borderJoinStyle: 'miter',
          pointBorderColor: 'rgba(75,192,192,1)',
          pointBackgroundColor: '#fff',
          pointBorderWidth: 1,
          pointHoverRadius: 5,
          pointHoverBackgroundColor: 'rgba(75,192,192,1)',
          pointHoverBorderColor: 'rgba(220,220,220,1)',
          pointHoverBorderWidth: 2,
          pointRadius: 1,
          pointHitRadius: 10,
        }
      ],
    }
  }

  chartMonthlyData = canvas => {
    const now = new Date();
    const { expencesSortedByMonth, months } = this.props;

    const ctx = canvas.getContext("2d")
    const gradient = ctx.createLinearGradient(0,10,1000,20);
    gradient.addColorStop(0, 'rgba(	23, 105, 170, 0.7)');
    gradient.addColorStop(1, 'rgba(	0, 150, 136, 0.7)');

    const borderGradient = ctx.createLinearGradient(0,10,1000,20);
    borderGradient.addColorStop(0, '#1769aa');
    borderGradient.addColorStop(1, '#009688');

    const expencesThisMonth = expencesSortedByMonth[months[now.getMonth()]][0]
    var options = { weekday: 'short', year: 'numeric', month: '2-digit', day: 'numeric' };

    const labels = expencesThisMonth.map((month, i) => {
      const date = new Date(expencesThisMonth[i].timestamp);
      return date.toLocaleString('en-us', options);
    })

    return {
      labels: labels,
      datasets: [
        {
          label: 'Amount',
          data: expencesThisMonth.map((month, i) => expencesThisMonth[i].amount),
          fill: true,
          backgroundColor: gradient,
          lineTension:0.0000001,
          borderColor: borderGradient,
          borderWidth: 5,
          borderCapStyle: 'butt',
          borderDash: [],
          borderDashOffset: 0.0,
          borderJoinStyle: 'miter',
          pointBorderColor: 'rgba(75,192,192,1)',
          pointBackgroundColor: '#fff',
          pointBorderWidth: 1,
          pointHoverRadius: 5,
          pointHoverBackgroundColor: 'rgba(75,192,192,1)',
          pointHoverBorderColor: 'rgba(220,220,220,1)',
          pointHoverBorderWidth: 2,
          pointRadius: 1,
          pointHitRadius: 10,
        }
      ],
    }
  }

  render() {
    const { expencesSortedByMonth } = this.props;

    return (
      <Paper style={{marginTop: '4em', marginBottom: '4em', padding: '2em'}}>
      <div className="center" style={{ marginBottom: '4em'}}>
      <h1 style={{fontWeight: '300'}}>History</h1>
      <Switch handleChange={this.handleChange}
        checkedA={this.state.checkedA}
        checkedB={this.state.checkedB}
        title1="This year"
        title2="This month"
      />
      <div className="center" style={{position: 'relative', height: '40vh', width: '80vw'}}>
      <Line
      data={this.state.checkedA ? this.chartYearlyData : this.chartMonthlyData}
      options={chartOptions}
      />
      </div>
      </div>
      <div>
      {
        Object.keys(expencesSortedByMonth).map((month, i) => {
          return <MonthlyExpencesExpantionPanel key={i +1} month={month} expences={expencesSortedByMonth[month]}/>
        })
      }
      </div>
      </Paper>
    )
  }
}

export default ExpenceHistory;
