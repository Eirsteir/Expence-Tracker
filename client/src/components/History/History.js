import React from "react";
import { Line } from "react-chartjs-2";

import "../../containers/Dashboard/Dashboard.css";

const chartOptions = currency => {
  return {
    maintainAspectRatio: false,
    scales: {
      yAxes: [
        {
          ticks: {
            suggestedMin: 10,
            suggestedMax: 100,
            fontColor: "rgba(255,255,255, 0.6)",
            fontWeight: 300,
            beginAtZero: true,
            maxTicksLimit: 5,
            padding: 20,
            // Include a dollar sign in the ticks
            callback: function(value, index, values) {
              return `${currency}${value}`;
            }
          },
          gridLines: {
            zeroLineColor: "transparent"
          },
          fontFamily: "Roboto"
        }
      ],
      xAxes: [
        {
          gridLines: {
            drawTicks: false,
            display: false
          },
          ticks: {
            padding: 20,
            fontColor: "rgba(255,255,255, 0.6)",
            fontWeight: 300
          }
        }
      ]
    },
    animation: {
      easing: "easeOutExpo"
    },
    legend: {
      display: false
    },
    tooltips: {
      callbacks: {
        label: function(tooltipItem) {
          return tooltipItem.yLabel;
        }
      }
    }
  };
};

const months = [
  "January",
  "Febuary",
  "March",
  "April",
  "May",
  "June",
  "July",
  "August",
  "September",
  "October",
  "November",
  "December"
];

class ExpenceHistory extends React.Component {
  chartData = canvas => {
    const now = new Date();
    const { expences } = this.props;

    const ctx = canvas.getContext("2d");
    const gradient = ctx.createLinearGradient(0, 0, 0, 170);
    gradient.addColorStop(0, "rgba(	204, 40, 93, 0.5)");
    gradient.addColorStop(1, "rgba(	0, 0, 0, 0.1)");

    const expencesThisMonth = expences[months[now.getMonth()]][0];
    var options = {
      month: "2-digit",
      day: "2-digit"
    };

    const labels = expencesThisMonth.map((month, i) => {
      const date = new Date(expencesThisMonth[i].timestamp);
      return date.toLocaleString("en-us", options);
    });

    const data = expencesThisMonth.map(
      (month, i) => expencesThisMonth[i].amount
    );

    return {
      labels: labels,
      datasets: [
        {
          label: "Remove this",
          data: data,
          fill: true,
          backgroundColor: gradient,
          lineTension: 0.35,
          borderColor: "rgba(204, 40, 93, 0.7)",
          borderWidth: 3,
          borderCapStyle: "butt",
          borderDash: [],
          borderDashOffset: 0.0,
          borderJoinStyle: "miter",

          pointHoverRadius: 5,
          pointHoverBackgroundColor: "rgba(204, 40, 93, 0.7)",
          pointHoverBorderColor: "rgba(204, 40, 93, 0.7)",
          pointHoverBorderWidth: 2,
          pointRadius: 1,
          pointHitRadius: 10
        }
      ]
    };
  };

  render() {
    return (
      <div
        style={{ padding: "1rem", backgroundColor: "#343b64", color: "#fff" }}
      >
        <div style={{ paddingBottom: "1.5rem" }}>Expences</div>
        <div className="center">
          <Line
            data={this.chartData}
            options={chartOptions(this.props.currency)}
            height={244}
          />
        </div>
      </div>
    );
  }
}

export default ExpenceHistory;
