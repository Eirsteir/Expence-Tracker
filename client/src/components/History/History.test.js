import { shallow } from "enzyme";
import React from "react";
import History from "./History.js";

it("expect to render CardList component", () => {
  const mockExpences = [
    { timestamp: new Date(0), tag: "Household", amount: 200 },
    { timestamp: new Date(0), tag: "Food", amount: 300 },
    { timestamp: new Date(0), tag: "Food", amount: 500 }
  ];
  expect(shallow(<History expences={mockExpences} />)).toMatchSnapshot();
});
