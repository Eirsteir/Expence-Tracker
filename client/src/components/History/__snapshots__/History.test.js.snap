// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CardList component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenceHistory
    expences={
        Array [
            Object {
              "amount": 200,
              "tag": "Household",
              "timestamp": 1970-01-01T00:00:00.000Z,
            },
            Object {
              "amount": 300,
              "tag": "Food",
              "timestamp": 1970-01-01T00:00:00.000Z,
            },
            Object {
              "amount": 500,
              "tag": "Food",
              "timestamp": 1970-01-01T00:00:00.000Z,
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="center"
        style={
                Object {
                        "marginBottom": "4em",
                      }
        }
>
        <h1
                style={
                        Object {
                                "fontWeight": "300",
                              }
                }
        >
                History
        </h1>
        <Switches
                checkedA={true}
                checkedB={false}
                handleChange={[Function]}
                title1="This year"
                title2="This month"
        />
        <div
                className="center"
                style={
                        Object {
                                "height": "60vh",
                                "position": "relative",
                                "width": "80vw",
                              }
                }
        >
                <Line
                        data={[Function]}
                        options={
                                Object {
                                        "animation": Object {
                                          "easing": "easeOutExpo",
                                        },
                                        "legend": Object {
                                          "position": "bottom",
                                        },
                                        "maintainAspectRatio": false,
                                        "scales": Object {
                                          "xAxes": Array [
                                            Object {
                                              "gridLines": Object {
                                                "zeroLineColor": "transparent",
                                              },
                                              "ticks": Object {
                                                "fontColor": "rgba(0,0,0,0.5)",
                                                "padding": 20,
                                              },
                                            },
                                          ],
                                          "yAxes": Array [
                                            Object {
                                              "fontFamily": "Roboto",
                                              "gridLines": Object {
                                                "display": false,
                                                "drawTicks": false,
                                              },
                                              "ticks": Object {
                                                "beginAtZero": true,
                                                "callback": [Function],
                                                "fontColor": "rgba(0,0,0,0.5)",
                                                "fontStyle": "bold",
                                                "maxTicksLimit": 5,
                                                "padding": 20,
                                                "suggestedMax": 100,
                                                "suggestedMin": 10,
                                              },
                                            },
                                          ],
                                        },
                                      }
                        }
                />
        </div>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            style={
                        Object {
                                    "fontWeight": "300",
                                  }
            }
>
            History
</h1>,
          <Switches
            checkedA={true}
            checkedB={false}
            handleChange={[Function]}
            title1="This year"
            title2="This month"
/>,
          <div
            className="center"
            style={
                        Object {
                                    "height": "60vh",
                                    "position": "relative",
                                    "width": "80vw",
                                  }
            }
>
            <Line
                        data={[Function]}
                        options={
                                    Object {
                                                "animation": Object {
                                                  "easing": "easeOutExpo",
                                                },
                                                "legend": Object {
                                                  "position": "bottom",
                                                },
                                                "maintainAspectRatio": false,
                                                "scales": Object {
                                                  "xAxes": Array [
                                                    Object {
                                                      "gridLines": Object {
                                                        "zeroLineColor": "transparent",
                                                      },
                                                      "ticks": Object {
                                                        "fontColor": "rgba(0,0,0,0.5)",
                                                        "padding": 20,
                                                      },
                                                    },
                                                  ],
                                                  "yAxes": Array [
                                                    Object {
                                                      "fontFamily": "Roboto",
                                                      "gridLines": Object {
                                                        "display": false,
                                                        "drawTicks": false,
                                                      },
                                                      "ticks": Object {
                                                        "beginAtZero": true,
                                                        "callback": [Function],
                                                        "fontColor": "rgba(0,0,0,0.5)",
                                                        "fontStyle": "bold",
                                                        "maxTicksLimit": 5,
                                                        "padding": 20,
                                                        "suggestedMax": 100,
                                                        "suggestedMin": 10,
                                                      },
                                                    },
                                                  ],
                                                },
                                              }
                        }
            />
</div>,
        ],
        "className": "center",
        "style": Object {
          "marginBottom": "4em",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "History",
            "style": Object {
              "fontWeight": "300",
            },
          },
          "ref": null,
          "rendered": "History",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "checkedA": true,
            "checkedB": false,
            "handleChange": [Function],
            "title1": "This year",
            "title2": "This month",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Line
              data={[Function]}
              options={
                            Object {
                                          "animation": Object {
                                            "easing": "easeOutExpo",
                                          },
                                          "legend": Object {
                                            "position": "bottom",
                                          },
                                          "maintainAspectRatio": false,
                                          "scales": Object {
                                            "xAxes": Array [
                                              Object {
                                                "gridLines": Object {
                                                  "zeroLineColor": "transparent",
                                                },
                                                "ticks": Object {
                                                  "fontColor": "rgba(0,0,0,0.5)",
                                                  "padding": 20,
                                                },
                                              },
                                            ],
                                            "yAxes": Array [
                                              Object {
                                                "fontFamily": "Roboto",
                                                "gridLines": Object {
                                                  "display": false,
                                                  "drawTicks": false,
                                                },
                                                "ticks": Object {
                                                  "beginAtZero": true,
                                                  "callback": [Function],
                                                  "fontColor": "rgba(0,0,0,0.5)",
                                                  "fontStyle": "bold",
                                                  "maxTicksLimit": 5,
                                                  "padding": 20,
                                                  "suggestedMax": 100,
                                                  "suggestedMin": 10,
                                                },
                                              },
                                            ],
                                          },
                                        }
              }
/>,
            "className": "center",
            "style": Object {
              "height": "60vh",
              "position": "relative",
              "width": "80vw",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": [Function],
              "options": Object {
                "animation": Object {
                  "easing": "easeOutExpo",
                },
                "legend": Object {
                  "position": "bottom",
                },
                "maintainAspectRatio": false,
                "scales": Object {
                  "xAxes": Array [
                    Object {
                      "gridLines": Object {
                        "zeroLineColor": "transparent",
                      },
                      "ticks": Object {
                        "fontColor": "rgba(0,0,0,0.5)",
                        "padding": 20,
                      },
                    },
                  ],
                  "yAxes": Array [
                    Object {
                      "fontFamily": "Roboto",
                      "gridLines": Object {
                        "display": false,
                        "drawTicks": false,
                      },
                      "ticks": Object {
                        "beginAtZero": true,
                        "callback": [Function],
                        "fontColor": "rgba(0,0,0,0.5)",
                        "fontStyle": "bold",
                        "maxTicksLimit": 5,
                        "padding": 20,
                        "suggestedMax": 100,
                        "suggestedMin": 10,
                      },
                    },
                  ],
                },
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="center"
          style={
                    Object {
                              "marginBottom": "4em",
                            }
          }
>
          <h1
                    style={
                              Object {
                                        "fontWeight": "300",
                                      }
                    }
          >
                    History
          </h1>
          <Switches
                    checkedA={true}
                    checkedB={false}
                    handleChange={[Function]}
                    title1="This year"
                    title2="This month"
          />
          <div
                    className="center"
                    style={
                              Object {
                                        "height": "60vh",
                                        "position": "relative",
                                        "width": "80vw",
                                      }
                    }
          >
                    <Line
                              data={[Function]}
                              options={
                                        Object {
                                                  "animation": Object {
                                                    "easing": "easeOutExpo",
                                                  },
                                                  "legend": Object {
                                                    "position": "bottom",
                                                  },
                                                  "maintainAspectRatio": false,
                                                  "scales": Object {
                                                    "xAxes": Array [
                                                      Object {
                                                        "gridLines": Object {
                                                          "zeroLineColor": "transparent",
                                                        },
                                                        "ticks": Object {
                                                          "fontColor": "rgba(0,0,0,0.5)",
                                                          "padding": 20,
                                                        },
                                                      },
                                                    ],
                                                    "yAxes": Array [
                                                      Object {
                                                        "fontFamily": "Roboto",
                                                        "gridLines": Object {
                                                          "display": false,
                                                          "drawTicks": false,
                                                        },
                                                        "ticks": Object {
                                                          "beginAtZero": true,
                                                          "callback": [Function],
                                                          "fontColor": "rgba(0,0,0,0.5)",
                                                          "fontStyle": "bold",
                                                          "maxTicksLimit": 5,
                                                          "padding": 20,
                                                          "suggestedMax": 100,
                                                          "suggestedMin": 10,
                                                        },
                                                      },
                                                    ],
                                                  },
                                                }
                              }
                    />
          </div>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              style={
                            Object {
                                          "fontWeight": "300",
                                        }
              }
>
              History
</h1>,
            <Switches
              checkedA={true}
              checkedB={false}
              handleChange={[Function]}
              title1="This year"
              title2="This month"
/>,
            <div
              className="center"
              style={
                            Object {
                                          "height": "60vh",
                                          "position": "relative",
                                          "width": "80vw",
                                        }
              }
>
              <Line
                            data={[Function]}
                            options={
                                          Object {
                                                        "animation": Object {
                                                          "easing": "easeOutExpo",
                                                        },
                                                        "legend": Object {
                                                          "position": "bottom",
                                                        },
                                                        "maintainAspectRatio": false,
                                                        "scales": Object {
                                                          "xAxes": Array [
                                                            Object {
                                                              "gridLines": Object {
                                                                "zeroLineColor": "transparent",
                                                              },
                                                              "ticks": Object {
                                                                "fontColor": "rgba(0,0,0,0.5)",
                                                                "padding": 20,
                                                              },
                                                            },
                                                          ],
                                                          "yAxes": Array [
                                                            Object {
                                                              "fontFamily": "Roboto",
                                                              "gridLines": Object {
                                                                "display": false,
                                                                "drawTicks": false,
                                                              },
                                                              "ticks": Object {
                                                                "beginAtZero": true,
                                                                "callback": [Function],
                                                                "fontColor": "rgba(0,0,0,0.5)",
                                                                "fontStyle": "bold",
                                                                "maxTicksLimit": 5,
                                                                "padding": 20,
                                                                "suggestedMax": 100,
                                                                "suggestedMin": 10,
                                                              },
                                                            },
                                                          ],
                                                        },
                                                      }
                            }
              />
</div>,
          ],
          "className": "center",
          "style": Object {
            "marginBottom": "4em",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "History",
              "style": Object {
                "fontWeight": "300",
              },
            },
            "ref": null,
            "rendered": "History",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "checkedA": true,
              "checkedB": false,
              "handleChange": [Function],
              "title1": "This year",
              "title2": "This month",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Line
                data={[Function]}
                options={
                                Object {
                                                "animation": Object {
                                                  "easing": "easeOutExpo",
                                                },
                                                "legend": Object {
                                                  "position": "bottom",
                                                },
                                                "maintainAspectRatio": false,
                                                "scales": Object {
                                                  "xAxes": Array [
                                                    Object {
                                                      "gridLines": Object {
                                                        "zeroLineColor": "transparent",
                                                      },
                                                      "ticks": Object {
                                                        "fontColor": "rgba(0,0,0,0.5)",
                                                        "padding": 20,
                                                      },
                                                    },
                                                  ],
                                                  "yAxes": Array [
                                                    Object {
                                                      "fontFamily": "Roboto",
                                                      "gridLines": Object {
                                                        "display": false,
                                                        "drawTicks": false,
                                                      },
                                                      "ticks": Object {
                                                        "beginAtZero": true,
                                                        "callback": [Function],
                                                        "fontColor": "rgba(0,0,0,0.5)",
                                                        "fontStyle": "bold",
                                                        "maxTicksLimit": 5,
                                                        "padding": 20,
                                                        "suggestedMax": 100,
                                                        "suggestedMin": 10,
                                                      },
                                                    },
                                                  ],
                                                },
                                              }
                }
/>,
              "className": "center",
              "style": Object {
                "height": "60vh",
                "position": "relative",
                "width": "80vw",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": [Function],
                "options": Object {
                  "animation": Object {
                    "easing": "easeOutExpo",
                  },
                  "legend": Object {
                    "position": "bottom",
                  },
                  "maintainAspectRatio": false,
                  "scales": Object {
                    "xAxes": Array [
                      Object {
                        "gridLines": Object {
                          "zeroLineColor": "transparent",
                        },
                        "ticks": Object {
                          "fontColor": "rgba(0,0,0,0.5)",
                          "padding": 20,
                        },
                      },
                    ],
                    "yAxes": Array [
                      Object {
                        "fontFamily": "Roboto",
                        "gridLines": Object {
                          "display": false,
                          "drawTicks": false,
                        },
                        "ticks": Object {
                          "beginAtZero": true,
                          "callback": [Function],
                          "fontColor": "rgba(0,0,0,0.5)",
                          "fontStyle": "bold",
                          "maxTicksLimit": 5,
                          "padding": 20,
                          "suggestedMax": 100,
                          "suggestedMin": 10,
                        },
                      },
                    ],
                  },
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
